### ISSUE TITLE
Attempted to call GetOrigin on invalid entity

### ISSUE THREAD

**Issue by fvnkhead** [Legonzaur/Northstar-HoloShift#6]
Server crashed with this error, not sure what caused it:

```
[09:37:18] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetOrigin on invalid entity
[09:37:18] [info] [SERVER SCRIPT]  -> initial_origin -= (initial_origin-decoy.GetOrigin())*(1/i)
[09:37:18] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [PlayerUsesHoloRewindThreaded()] weapons/mp_ability_holopilot.nut line [456]

[09:37:18] [info] [SERVER SCRIPT] LOCALS
[i] 7
[positions] ARRAY
[initial_angle] <0, -163.477, 0>
[initial_origin] <227.25, -71.9688, 356.594>
[mover] ENTITY (script_mover_lightweight [142] at <227.25 -71.9688 356.594>)
[decoy] ENTITY (NULL)
[player] ENTITY (player puraI_gt [3] (player "puraI_gt" at <227.25 -71.9688 356.594>))
[this] TABLE

DIAGPRINTS

[09:37:18] [info] [SERVER SCRIPT] SCRIPT ERROR: [SERVER] Attempted to call GetOrigin on invalid entity
[09:37:18] [info] [SERVER SCRIPT]  -> player.SetOrigin(decoy.GetOrigin())
[09:37:18] [info] [SERVER SCRIPT]
CALLSTACK
*FUNCTION [unknown()] weapons/mp_ability_holopilot.nut line [433]
*FUNCTION [SpinOff()] NATIVE line [-1]
*FUNCTION [PlayerUsesHoloRewind()] weapons/mp_ability_holopilot.nut line [419]
*FUNCTION [OnWeaponPrimaryAttack_holopilot()] weapons/mp_ability_holopilot.nut line [141]

[09:37:18] [info] [SERVER SCRIPT] LOCALS
[player] ENTITY (player puraI_gt [3] (player "puraI_gt" at <227.25 -71.9688 356.594>))
[mover] ENTITY (script_mover_lightweight [142] at <227.25 -71.9688 356.594>)
[decoy] ENTITY (NULL)
[this] TABLE
[decoy] ENTITY (NULL)
[player] ENTITY (player puraI_gt [3] (player "puraI_gt" at <227.25 -71.9688 356.594>))
[this] TABLE
[decoy] ENTITY (NULL)
[weaponOwner] ENTITY (player puraI_gt [3] (player "puraI_gt" at <227.25 -71.9688 356.594>))
[attackParams] unknown struct
[weapon] ENTITY (weaponx mp_ability_holopilot [147] (weapon "mp_ability_holopilot" at <2095.18 -167.46 64.7954>))
[this] TABLE

DIAGPRINTS

0024:fixme:msvcp:_Locinfo__Locinfo_ctor_cat_cstr (00000000007CE1A0 1 C) semi-stub
0024:fixme:msvcp:_Locinfo__Locinfo_ctor_cat_cstr (00000000007CDEF0 1 C) semi-stub
[09:37:18] [error] Northstar has crashed! a minidump has been written and exception info is available below:
[09:37:18] [error] Cause: Access Violation
Attempted to read from: 0x0000000000000000
```

**Comment by fvnkhead** [comment]
More info: we were playing Marked For Death and apparently this happened when someone got marked while phasing

**Comment by Legonzaur** [comment]
Maybe I should check if decoy is valid before teleporting player on thread end

```c

	OnThreadEnd( function() : ( player, mover, decoy )
	{
		if(IsValid(decoy)){
			player.SetOrigin(decoy.GetOrigin())
		        player.SetAngles(decoy.GetAngles())
		        player.SetVelocity(decoy.GetVelocity())
		}
		CancelPhaseShift( player )
		player.DeployWeapon()
		player.SetPredictionEnabled( true )
		player.ClearParent()
		ViewConeFree( player )
		mover.Destroy()
		CleanupExistingDecoy(decoy)
	})

```

**Comment by Legonzaur** [comment]
Should have been fixed with #7
